verticaltabs.jsm

```diff
@@ -339,6 +340,38 @@ VerticalTabs.prototype = {
     });
     toolbar.appendChild(pin_button);
     leftbox.insertBefore(toolbar, leftbox.firstChild);
+    let find_input = this.createElement('textbox', {
+      'id': 'find-input',
+      'class': 'searchbar-textbox'
+    });
+    let search_icon = this.createElement('image', {
+      'id': 'tabs-search'
+    });
+    find_input.appendChild(search_icon);
+    find_input.addEventListener('input', this.filtertabs.bind(this));
+    this.window.addEventListener('keyup', function (e){
+      if(e.keyCode === 27) {
+        this.clearFind();
+      }
+    }.bind(this));
+    document.getElementById('filler-tab').addEventListener('click', this.clearFind.bind(this));
+
+    let spacer = this.createElement('spacer', {'id': 'new-tab-spacer'});
+    toolbar.insertBefore(find_input, pin_button);
+    toolbar.insertBefore(spacer, pin_button);
+
+    function resizeFindInput() {
+      if (this.pinnedWidth > 190) {
+        spacer.style.visibility = 'collapse';
+        find_input.style.visibility = 'visible';
+      } else {
+        find_input.style.visibility = 'collapse';
+        spacer.style.visibility = 'visible';
+      }
+    }
+
+    resizeFindInput.bind(this)();
```

Comment:
I don't think we need the `.bind(this)` here, since we're calling it in this already…  (That also applies to line 298.)